src/java/org/apache/cassandra/security/PEMBasedSslContextFactory.java

```diff
@@ -241,6 +262,39 @@ protected KeyManagerFactory buildKeyManagerFactory() throws SSLException
         }
     }
 
+    @Override
+    protected KeyManagerFactory buildOutboundKeyManagerFactory() throws SSLException {
+        try
+        {
+            if (hasOutboundKeystore())
+            {
+                if (maybeFileBasedOutboundPrivateKey)
+                {
+                    pemEncodedOutboundKey = readPEMFile(outboundKeystore); // read PEM from the file
+                }
+
+                KeyManagerFactory kmf = KeyManagerFactory.getInstance(
+                algorithm == null ? KeyManagerFactory.getDefaultAlgorithm() : algorithm);
+                KeyStore ks = buildKeyStore(pemEncodedOutboundKey, outboundKeyPassword);
+                if (!checkedExpiry)
```

Comment:
@jyothsnakonisa I think we should should separate the cert expiry checks for inbound vs outbound keys. According to this code once we verified client (or server) key certs it will skip the other. We should have `outboundCertExpiryCheck` sort of a flag on top of the existing one.

Reply:
Thanks @maulin-vasavada, I addressed this issue in the following commit.